python/lsst/ap/verify/measurements/compute_metrics.py

```diff
@@ -62,4 +74,84 @@ def measure_from_metadata(metadata):
         if measurement is not None:
             result.append(measurement)
 
+    measurement = measure_number_new_dia_objects(
+        metadata, 'association', 'association.numNewDiaObjects')
+    if measurement is not None:
+        result.append(measurement)
+    measurement = measure_fraction_updated_dia_objects(
+        metadata, 'association', 'association.fracUpdatedDiaObjects')
+    if measurement is not None:
+        result.append(measurement)
+    measurement = measure_number_unassociated_dia_objects(
+        metadata, 'association', 'association.numUnassociatedDiaObjects')
+    if measurement is not None:
+        result.append(measurement)
+    return result
+
+
+def measure_from_butler_repo(repo, dataId):
+    """ Create measurements from a butler repository.
+
+    Parameters
+    ----------
+    repo: `str`
+        The output repository location to read from disk.
+    dataId: `str`
+        Butler identifier naming the data to be processed (e.g., visit and
+        ccdnum) formatted in the usual way (e.g., 'visit=54321 ccdnum=7').
+
+    Returns
+    -------
+    a list of `lsst.verify.Measurement` derived from `metadata`. May be empty.
+    """
+    result = []
+
+    dataId_items = re.split('[ +=]', dataId)
+    dataId_dict = dict(zip(dataId_items[::2], dataId_items[1::2]))
+    # Unfortunately this currently hard codes these measurements to be
+    # from one ccd/visit and requires them to be from DECam because
+    # of ccdnum. Buttler.get appears to require that visit and ccdnum
+    # both be ints rather than allowing them to be string type.
+    if 'visit' not in dataId_dict.keys():
+        raise RuntimeError('The dataId string is missing \'visit\'')
+    else:
+        visit = int(dataId_dict['visit'])
+        dataId_dict['visit'] = visit
+    if 'ccdnum' not in dataId_dict.keys():
+        raise RuntimeError('The dataId string is missing \'ccdnum\'')
+    else:
+        ccdnum = int(dataId_dict['ccdnum'])
+        dataId_dict['ccdnum'] = ccdnum 
+
+    butler = dafPersist.Butler(repo)
+    measurement = measure_number_sci_sources(
+        butler, dataId_dict, "ip_diffim.numSciSources")
+    if measurement is not None:
+        result.append(measurement)
+
+    measurement = measure_fraction_dia_sources_to_sci_sources(
+        butler, dataId_dict, "ip_diffim.fracDiaSourcesToSciSources")
+    if measurement is not None:
+        result.append(measurement)
+    return result
+
+
+def measure_from_L1_db_sqlite(db_name):
+    """ Make measurements on a sqlite database containing associated DIAObjects
```

Comment:
Based on the code below, do you mean "unassociated DIAObjects"? If not, maybe add a comment in the code below explaining the measurement further.

Reply:
The idea here is that this is a general use method for computing any measurements that need to access the data products in the association database. I've changed the docstring to this "Make measurements on a sqlite database containing the results of from source association."